src/plugins/analysis/information_leaks/code/information_leaks.py

```diff
@@ -0,0 +1,91 @@
+              }
+
+PATH_ARTIFACT_DICT = {
+    '.git|config': 'git_repo',
+    '.conda|environments.txt': 'conda_environment',
+    'default.conf': 'possible_code_blocks_config',
+    'clion64.exe.vmoptions': 'clion_jvm_options',
+    'idea.properties': 'clion_platform_properties',
+    '.config|Code|User|settings.json': 'vscode_settings',
+
+    '.cproject': 'eclipse_config',
+    '.csproject': 'eclipse_config',
+    '.project': 'eclipse_config',
+
+    '.hws': 'renesas_project_config',
+    '.ewd': 'iar_embedded_workbench_config',
+    '.ewp': 'iar_embedded_workbench_config',
+    '.eww': 'iar_embedded_workbench_config',
+    '.ewt': 'iar_embedded_workbench_config',
+
+    '.Uv2': 'keil_uvision_config',
+    '.uvproj': 'keil_uvision_config',
+    '.uvopt': 'keil_uvision_config',
+    '.uvprojx': 'keil_uvision_config',
+    '.uvoptx': 'keil_uvision_config',
+
+    '.atsln': 'atmel_studio_config',
+    '.cyprj': 'cydesigner_config',
+    '.cywrk': 'cydesigner_config'
+}
+
+DIRECTORY_DICT = {
+    '.git': 'git_config_repo',
+    '.github': 'github_config_directory',
+    '.pytest_cache': 'pytest_cache_directory',
+    '.conda': 'conda_directory',
+    '.config': 'a_config_directory',
+    '.subversion': 'svn_user_settings_directory',
+    'subversion': 'svn_settings_directory',
+    '.idea': 'pycharm_config_directory'
+}
+
+
+class AnalysisPlugin(AnalysisBasePlugin):
+    """
+    This Plugin searches for compilation artifacts in a firmware,
+        e.g., github repositories, IDE configs and special paths
+    """
+    NAME = 'information_leaks'
+    DEPENDENCIES = []
+    DESCRIPTION = 'Find leaked information like compilation artifacts'
+    MIME_WHITELIST = ['application/x-executable', 'application/x-object', 'application/x-sharedlib', 'text/plain']
+    VERSION = '0.1'
+
+    def __init__(self, plugin_administrator, config=None, recursive=True, offline_testing=False):
+        super().__init__(plugin_administrator, config=config, recursive=recursive, plugin_path=__file__,
+                         offline_testing=offline_testing)
+
+    def process_object(self, file_object: FileObject):
+        if file_object.processed_analysis['file_type']['mime'] == 'text/plain':
+            self._find_artifacts(file_object)
+        else:
+            for label, regex in PATH_REGEX.items():
+                self._find_paths(file_object, regex, label)
+        file_object.processed_analysis[self.NAME]['summary'] = list(file_object.processed_analysis[self.NAME])
+        return file_object
+
+    def _find_artifacts(self, file_object: FileObject):
+        for virtual_path_list in file_object.virtual_file_path.values():
+            for virtual_path in virtual_path_list:
+                for key_path, artifact in PATH_ARTIFACT_DICT.items():
+                    if virtual_path.endswith(key_path):
+                        file_object.processed_analysis[self.NAME][artifact] = file_object.binary.decode()
```

Comment:
After testing it again, I have to reconsider: It probably isn't a good idea to simple save the complete file content in the analysis result (and therefore in the database). If the user wants to see the content. he can do that using the preview functionality.

Instead, it is probably better to only store 
```suggestion
                        file_object.processed_analysis[self.NAME]['path_artifacts'] = artifact
```
Sadly, this makes our previous changes to the summary generation obsolete. We still want the `artifact`s in the summary.

Reply:
Why would it make it obsolete? We would just store the file paths, just like we have done with directories or am I missing something?

Reply:
What I meant was that we were able to use common approach for generating the summary for all cases but now the structure is different in this case if we want to keep `artifact` as summary entry. Therefore, each part again needs its own summary generation.